.changeset/ten-cougars-pretend.md

```diff
@@ -0,0 +1,85 @@
+---
+'@astrojs/netlify': major
+---
+
+# Netlify Adapter v4 simplifies static + SSR deployments
+
+This update is a complete overhaul of the Netlify adapter.
+It simplifies the user-facing config, and resolves a number of bugs along the way.
+
+Here's what changes:
+
+## Middleware without configuration
+
+In v3, [Middleware](https://docs.astro.build/en/guides/middleware/) was deployed to the SSR Function by default.
+Enabling the `edgeMiddleware` option would change this behaviour to deploy it using Netlify Edge Functions instead, improving
+page speed and enabling the middleware to run on prerendered sites.
+
+In v4, this is the default setting.
+If your Astro site makes use of Middleware,
+it's deployed to an Edge Function where it runs super close to your users and is also applied to static sites.
+You can use this to implement customization or authorization logic!
+
+**Action Required:**
+If your site is using Middleware, and you had `edgeMiddleware` enabled, nothing changes for you - feel free to remove the config field, though.
+If your site is using Middleware, and you didn't have `edgeMiddleware` enabled, your Middleware will now be deployed to an Edge Function (probably no change needed) and start running on prerendered pages (check if this requires any changes to your middleware logic).
```

Comment:
> it would be nice to say something as to how they can check it?

They should look at all prerendered pages, and check if they still work as intended now that the middleware applies to them. Will add another line about that 👍

Reply:
how do you like https://github.com/Skn0tt/adapters/pull/1/commits/f52963b00fcb7d18e5facc073232c3bd0c2f406b?